android-app-examples/SNetDemo/app/src/main/java/com/example/snetdemo/ImageSegmentationActivity.java

```diff
@@ -0,0 +1,449 @@
+    private Button btn_RunImageSegmentation;
+    private Button btn_GrabCameraImage;
+
+    private ImageView imv_Input;
+
+    private TextView textViewProgress;
+    private TextView textViewResponseTime;
+
+    private Bitmap mDecodedBitmap = null;
+
+    private RelativeLayout loadingPanel;
+
+    private long mServiceResponseTime = 0;
+    private boolean mIsDeviceWithCamera = true;
+
+    String mCurrentPhotoPath = "";
+    Uri mCameraImageURI;
+
+    byte[] mBytesInput = null;
+
+    String mImageInputPath = null;
+    String mImageSegmentedPath = null;
+
+    private String mErrorMessage = "";
+    private boolean isExceptionCaught = false;
+
+    private int channelID;
+    private SemanticSegmentationService semanticSegmentationService;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_image_segmentation);
+
+        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+
+            requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1);
+        }
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.INTERNET)
+                != PackageManager.PERMISSION_GRANTED) {
+
+            requestPermissions(new String[]{Manifest.permission.INTERNET}, 1);
+        }
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.CAMERA}, 1);
+        }
+
+        setTitle("Image Segmentation Demo");
+
+        btn_UploadImageInput = (Button) findViewById(R.id.btn_uploadImageForSegmentation);
+        btn_RunImageSegmentation = (Button) findViewById(R.id.btn_runImageSegmentation);
+        btn_RunImageSegmentation.setEnabled(false);
+
+        btn_GrabCameraImage = (Button) findViewById(R.id.btn_grabCameraImage);
+
+        imv_Input = (ImageView)findViewById(R.id.imageViewInput);
+
+        loadingPanel = (RelativeLayout) findViewById(R.id.loadingPanel);
+        loadingPanel.setVisibility(View.INVISIBLE);
+
+        textViewProgress = findViewById(R.id.textViewProgress);
+        textViewProgress.setVisibility(View.INVISIBLE);
+
+        textViewResponseTime = findViewById(R.id.textViewResponseTime);
+        textViewResponseTime.setText("Service response time (ms):");
+        textViewResponseTime.setVisibility(View.INVISIBLE);
+
+
+        PackageManager pm = this.getPackageManager();
+        if (!pm.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY))
+        {
+            mIsDeviceWithCamera = false;
+            btn_GrabCameraImage.setEnabled(false);
+        }
+
+        channelID = this.getResources().getInteger(R.integer.channel_id);
```

Comment:
I would add comment for users who try to run and gets exception here.
```suggestion
        // See README.md on how to initialize channel ID via resources
        channelID = this.getResources().getInteger(R.integer.channel_id);
```